### DISCUSSION TITLE
What are the limitations of config.json compared to "direct python" ?

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#2037] [General Q&A] [answered]
Hi,

THere are two ways of running the training one via config.json the other one via so called direct python.

The config.json way is called "old way" in the documentation. So if it is old, is it deprecated ? Why did you move to the direct python, what were the limitations of the config.json ?

I am asking because I am used to config.json, but sometimes the training runs fine sometimes not and the setting in config.json are not always taken into account. So I am wondering whether I am missing something by not using the direct python way.

Thank you in advance for the clarifications!

**Comment by king-dahmanus** [comment]
hey there. Is the direct python way the one where you edit the parameter from the recipe training file before running it? Or is it some sort of comand line argument business

**Reply by Ca-ressemble-a-du-fake** [reply]
Yes it is!

**Comment by king-dahmanus** [comment]
it is what? You didn't specify between the two, sir

**Reply by Ca-ressemble-a-du-fake** [reply]
😄 You asked :
"Is the direct python way the one where you edit the parameter from the recipe training file before running it?" 
That's what I answered positively !

**Comment by erogol** [comment, marked as answer]
Python way is more flexible and mostly I run my experiments like that. But @Edresson prefers config.json for his reasons :)

**Reply by Ca-ressemble-a-du-fake** [reply]
Ok got it thank you!
